New module for querying ACM certificates

willthames opened the pull request:
##### SUMMARY

Provide a facts module for AWS' ACM service

Note: not using AnsibleAWSModule so that it can be dropped into
a module library of Ansible 2.3

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME
aws_acm_facts

##### ANSIBLE VERSION
```
ansible 2.5.0 (devel 0c291ece1a) last updated 2017/09/11 10:16:27 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/will/src/ansible/lib/ansible
  executable location = /home/will/src/ansible/bin/ansible
  python version = 2.7.13 (default, Sep  5 2017, 08:53:59) [GCC 7.1.1 20170622 (Red Hat 7.1.1-3)]
```

s-hertel commented on lib/ansible/modules/cloud/amazon/aws_acm_facts.py:
```diff
@@ -0,0 +1,342 @@
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        client = boto3_conn(module, conn_type='client', resource='acm',
+                            region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.NoCredentialsError as e:
```
I think this can raise botocore.exceptions.ProfileNotFound too.

s-hertel commented on lib/ansible/modules/cloud/amazon/aws_acm_facts.py:
```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/python
+# This file is part of Ansible
```
New copyright: http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#copyright

Commit:
New module for querying ACM certificates

Not using AnsibleAWSModule so that it can be dropped into
a module library of Ansible 2.3

willthames commented:
@s-hertel thanks, I've addressed your feedback.

s-hertel commented:
shipit

gunzy83 reviewed (approved):
shipit
